src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/srm/algebras/BDD.cs

```diff
@@ -194,90 +193,90 @@ public BDD[] TopSort()
                 {
-                    nodes.AddRange(nonterminals[i]);
+                    sorted.AddRange(nonterminals[i]);
                 }
             }
 
-            return nodes.ToArray();
+            return sorted.ToArray();
         }
 
         #region Serialization
-        private static readonly long[] s_False_repr = new long[] { 0 };
-        private static readonly long[] s_True_repr = new long[] { 1 };
-
         /// <summary>
-        /// Serialize this BDD in a flat ulong array.
-        /// The BDD may have at most 2^k ordinals and 2^n nodes, st. k+2n&lt;64.
+        /// Serialize this BDD in a flat ulong array. The BDD may have at most 2^k ordinals and 2^n nodes, such that k+2n &lt; 64
         /// BDD.False is represented by return value ulong[]{0}.
         /// BDD.True is represented by return value ulong[]{1}.
-        /// Serializer uses more compacted representations when fewer bits are needed, which
-        /// is reflected in the first two numbers of the return value.
-        /// MTBDD terminals are represented by negated numbers as -id.
+        /// Serializer uses more compacted representations when fewer bits are needed, which is reflected in the first
+        /// two numbers of the return value. MTBDD terminals are represented by negated numbers as -id.
         /// </summary>
         public long[] Serialize()
         {
             if (IsEmpty)
-                return s_False_repr; //represents False
+                return s_falseRepresentation;
 
             if (IsFull)
-                return s_True_repr; //represents True
+                return s_trueRepresentation;
 
             if (IsLeaf)
                 return new long[] { 0, 0, -Ordinal };
 
-            BDD[] nodes = TopSort();
+            BDD[] nodes = TopologicalSort();
 
             Debug.Assert(nodes[nodes.Length - 1] == this);
             Debug.Assert(nodes.Length <= (1 << 24));
 
-            //use fewer bits when possible, starting with nibble size
+            // As few bits as possible are used to for ordinals and node identifiers for compact serialization.
+            // use at least a nibble (4 bits) to represent the ordinal and count how many are needed
             int ordinal_bits = 4;
             while (Ordinal >= (1 << ordinal_bits))
             {
                 ordinal_bits += 1;
             }
 
-            //use as few bits as possible for node identifiers, starting with 2 bits
-            //this will give smaller and more compact serialized representations
+            // use at least 2 bits to represent the node identifier and count how many are needed
             int node_bits = 2;
             while (nodes.Length >= (1 << node_bits))
             {
                 node_bits += 1;
             }
 
-            //add 2 extra positions: index 0 and 1 are reserved for False and True
+            // reserve space for all nodes plus 2 extra: index 0 and 1 are reserved for False and True
```

Comment:
```suggestion
            // Reserve space for all nodes plus 2 extra: index 0 and 1 are reserved for False and True
```

Reply:
To clarify, is the style to use for comments "always capitalized, but lone single-line comments without a period"?

Reply:
Space between the // and the start and starting with a captial.

We're inconsistent about whether things end with a period or not.

As for line length, we're also inconsistent, though for borderline cases we typically err on the side of longer comment lines rather than more lines.  I've not counted our average line length, but it's probably a bit more than the 80 of yesteryear :)

Reply:
Great, thanks! I set my editor to wrap at 120, which should fit into GitHub code views.